libs/core/src/lib/multi-input/multi-input-mobile/multi-input-mobile.component.ts

```diff
@@ -0,0 +1,158 @@
+    private _dialogRef: DialogRef;
+    /** @hidden */
+    private _selectedBackup: any[];
+
+    constructor(
+        @Optional() private dialogService: DialogService,
+        @Optional() private _multiInputComponent: MultiInputComponent,
+        private _changeDetRef: ChangeDetectorRef,
+        private _elementRef: ElementRef
+    ) {}
+
+    /** @hidden */
+    ngOnInit(): void {
+        this.multiInputConfig = this.getMultiInputConfig();
+        if (this.multiInputConfig) {
+            this._listenOnMultiInputOpenChange();
+        }
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        if (!this.dialogService && isDevMode()) {
+            throw new Error('There is no dialog service provided. Multi input can\'t be opened');
+        }
+        this._overwriteDialogProperties();
+        this._open();
+        this._dialogRef.hide(true);
+    }
+
+    /**
+     * Method returns, if the dialog is open.
+     * It gives information to prevent from having more than 1 dialog .
+     * Also removes from DOM, <input> control element, which is rendered in dialog.
+     */
+    hasOpenDialogs(): boolean {
+        return this.dialogService && this.dialogService.hasOpenDialogs();
+    }
+
+    /** Throw select all event, it's handled by multi input component */
+    selectAll(): void {
+        this._multiInputComponent.selectAllItems();
+    }
+
+    /** @hidden */
+    public handleDismiss(): void {
+        this._dialogRef.hide(true);
+        this._multiInputComponent.dialogDismiss(this._selectedBackup);
+    }
+
+    /** @hidden */
+    public handleApprove(): void {
+        this._dialogRef.hide(true);
+        this._multiInputComponent.dialogApprove();
+    }
+
+    /** @hidden */
+    private _overwriteDialogProperties(): void {
+        const multiInputConfig = this.getMultiInputConfig();
+        if (!multiInputConfig) {
+            return;
+        }
+
+        if (!this._multiInputComponent.dialogConfig) {
+            this._multiInputComponent.dialogConfig = new DialogConfig();
+        }
+    }
+
+    /** @hidden */
+    private getMultiInputConfig(): MultiInputMobileConfiguration {
+        if (this._multiInputComponent.multiInputMobileConfig) {
+            return this._multiInputComponent.multiInputMobileConfig;
+        } else if (this._multiInputComponent.providedMultiInputConfig) {
+            return this._multiInputComponent.providedMultiInputConfig;
+        } else {
+            if (isDevMode()) {
+                throw new Error('There is no multi input configuration object provided. ' +
+                    'You need to pass it as a "[multiInputMobileConfig]",' +
+                    'or provide it with "MULTI_INPUT_MOBILE_CONFIG" injection token'
+                );
+            }
+        }
+    }
+
+    private _toggleDialog(open: boolean): void {
+        if (open) {
+            this._selectedBackup = [...this._multiInputComponent.selected];
+            if (!this.hasOpenDialogs()) {
```

Comment:
Actually I encountered some issues. Dialog was deleted, after backdrop had been clicked. After setting closeOnBackdrop to false, it works fine. But still if something goes wrong, It will just re-create dialog :D